# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/PWebGL.ipdl
# Commit: b9f43d8a6c2d
# Full Hash: b9f43d8a6c2d9504dde533d3db13111ae4d31d06
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-07-12 21:56:41
# Regressor Bug: 1777426
# File Overlap Count: 2
# Description:
#   Bug 1777426 - Support fast readbacks in CopyToSwapChain with async present. r=sotaro,jgilbert
#   
#   Currently CopyToSwapChain creates spurious copies of the back buffer when SharedSurfaces aren't exportable (= ToSurfaceDescriptor returns Nothing from SharedSurface_Basic). These then later get read back into a CPU memory buffer when PresentFrontBufferToCompositor is used to send the buffer to RemoteTextureMap. This has associated performance and memory costs.
#   
#   Conceptually, we want Present/CopyToSwapChain to just do the right thing and automatically push buffers to RemoteTextureMap, rather than secondarily needing a hidden call to PresentFrontBufferToCompositor. Then we can get rid of the need to create front buffers whose only purpose is to shuttle data to PresentFrontBufferToCompositor which then shuttles it RemoteTextureMap.
# ==============================================================================

diff -r 0a598b1533e6 -r b9f43d8a6c2d dom/canvas/PWebGL.ipdl
--- a/dom/canvas/PWebGL.ipdl	Tue Jul 12 06:49:11 2022 +0000
+++ b/dom/canvas/PWebGL.ipdl	Tue Jul 12 06:56:19 2022 +0000
@@ -9,8 +9,6 @@
 include "mozilla/layers/LayersMessageUtils.h";
 
 using mozilla::layers::CompositableHandle from "mozilla/layers/LayersTypes.h";
-using mozilla::layers::RemoteTextureId from "mozilla/layers/LayersTypes.h";
-using mozilla::layers::RemoteTextureOwnerId from "mozilla/layers/LayersTypes.h";
 using mozilla::layers::SurfaceDescriptor from "mozilla/layers/LayersTypes.h";
 using std::string from "string";
 using mozilla::uvec2 from "mozilla/dom/WebGLIpdl.h";
@@ -76,7 +74,6 @@
                                                   uint32_t attachment,
                                                   uint32_t pname) returns (double? ret);
   sync GetFrontBuffer(uint64_t fb, bool vr) returns (SurfaceDescriptor? ret);
-  async PresentFrontBufferToCompositor(uint64_t fb, RemoteTextureId textureId, RemoteTextureOwnerId ownerId);
   sync GetIndexedParameter(uint32_t target, uint32_t index) returns (double? ret);
   sync GetInternalformatParameter(uint32_t target, uint32_t internalFormat, uint32_t pname) returns (Int32Vector? ret);
   sync GetLinkResult(uint64_t id) returns (LinkResult ret);